sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,114 @@
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.process.Props;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public final class Cluster {
+  private static final Logger LOG = LoggerFactory.getLogger(Cluster.class);
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  private boolean active = false;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param props the props
+   */
+  protected Cluster(Props props) {
+    ClusterProperties.completeDefaults(props);
+
+    if (props.valueAsBoolean(ClusterProperties.ENABLED)) {
+      initializeHazelcast(props);
+    }
+  }
+
+  private void initializeHazelcast(Props props) {
+    Config hzConfig = new Config();
+    // Configure the network instance
+    NetworkConfig netConfig = hzConfig.getNetworkConfig();
+    netConfig.setPort(props.valueAsInt(ClusterProperties.PORT))
+      .setPortAutoIncrement(props.valueAsBoolean(ClusterProperties.PORT_AUTOINCREMENT));
+
+    if (props.value(ClusterProperties.INTERFACES) != "") {
+      netConfig.getInterfaces()
+        .setEnabled(true)
+        .setInterfaces(
+          Arrays.asList(props.value(ClusterProperties.INTERFACES).split(","))
+        );
+    }
+
+    // Only allowing TCP/IP configuration
+    JoinConfig joinConfig = netConfig.getJoin();
+    joinConfig.getAwsConfig().setEnabled(false);
+    joinConfig.getMulticastConfig().setEnabled(false);
+    joinConfig.getTcpIpConfig().setEnabled(true);
+
+    // Add the defined members of the cluster
+    if (props.value(ClusterProperties.MEMBERS) != "") {
+      for (String member : props.value(ClusterProperties.MEMBERS).split(",")) {
+        if (!member.contains(":")) {
+          // If member does not define his port, let's use our default one
+          member += ":" + ClusterProperties.DEFAULT_PORT;
+        }
+        joinConfig.getTcpIpConfig().addMember(member);
+      }
+    }
+
+    // Tweak HazelCast configuration
+    hzConfig
+      .setProperty("hazelcast.tcp.join.port.try.count", "10") // Increase the number of tries
+      .setProperty("hazelcast.socket.bind.any", "false") // Don't bind all interfaces
+      .setProperty("hazelcast.phone.home.enabled", "false") // Don't phone home
+      .setProperty("hazelcast.logging.type", "slf4j"); // Use slf4j for logging
+
+    // We are not using the partition group of Hazelcast, so disabling it
+    hzConfig.getPartitionGroupConfig().setEnabled(false);
+
+    hazelcastInstance = Hazelcast.newHazelcastInstance(hzConfig);
+    active = true;
+  }
+
+  /**
+   * Is the cluster active
```

Comment:
I think description and naming is incorrect.

1. according to implementation, this flag indicates that the `hazelcastInstance` has been created. We could just as well test the `hazelcastInstance` property to know that
2. to really test that Hazelcast is up, I believe it would be more appropriate to rely on some Hazelcast's status API to compute this information. Here we only know we created the `HazelcastInstance` object.